### DISCUSSION TITLE
Setup instructions for non-tech IndieHacker

### DISCUSSION THREAD

**Discussion by adamwintle** [AntonOsika/gpt-engineer#209] [General]
Hi, I'm a non-technical IndieHacker, usually using no-code tools, which always have their limitations. I'd absolutely love to use GPT Engineer as my development team, however I'm struggling to set it up on my own server, a GutHub Codespaces or my local host. 

Would it be possible to get a step by step guide for totally non-technical people? 

Also, has anyone seen a demo of this running on a domain with a "input your own API key option"?

**Comment by noxiouscardiumdimidium** [comment]
go to folder you want the gptengineer folder to be in, open terminal/cmd from that location . git clone main repo with `git clone https://github.com/AntonOsika/gpt-engineer.git` run `pip install gpt-engineer` to get major update from auto install version. then `git pull` to get the great improvement commits from the last few days (the past 2 days updates: GUCCI!") if it fails to finish install because of dependencies, look near the bottom of the error, a lot times, especially with python, it litteraly tells you exactly how to fix the issue... like if it crashes because it missing npm, and the last line says "run pip install npm to fix this issue" guess what the ONLY thing that's going to fix it is... xD also: some of these major packages cant load and be accessed by the session that installed then... sometimes the solution is simply closing the terminal, and launching a new one so it can load can load dependencies that are blocked by running processes...
